packages/core/addon/components/navi-visualizations/goal-gauge.ts

```diff
@@ -9,93 +9,84 @@
  *  }}
  */
 
-import { alias } from '@ember/object/computed';
-import Component from '@ember/component';
-import { computed } from '@ember/object';
+import { alias, readOnly } from '@ember/object/computed';
+import Component from '@glimmer/component';
+import { computed, action } from '@ember/object';
 import { guidFor } from '@ember/object/internals';
-import { inject as service } from '@ember/service';
-import numeral from 'numeral';
+import { assert } from '@ember/debug';
+// @ts-ignore
 import d3 from 'd3';
-import layout from '../../templates/components/navi-visualizations/goal-gauge';
-import { canonicalizeMetric } from 'navi-data/utils/metric';
-import { layout as templateLayout, tagName } from '@ember-decorators/component';
+import numeral from 'numeral';
+import FilterFragment from 'navi-core/models/bard-request-v2/fragments/filter';
+import { VisualizationModel } from './table';
+import { GoalGaugeConfig } from 'navi-core/models/goal-gauge';
+import ColumnFragment from 'navi-core/models/bard-request-v2/fragments/column';
 
 const DEFAULT_OPTIONS = {
   baselineValue: 0,
   goalValue: 0,
   metricTitle: '',
-  metric: { metric: '', parameters: {} },
+  metricCid: '',
   prefix: '',
   unit: '',
   thresholdColors: ['#f05050', '#ffc831', '#44b876'],
   thresholdPercentages: [75, 85, 100]
 };
+export type Args = {
+  model: VisualizationModel;
+  options: GoalGaugeConfig['metadata'];
+};
 
-@templateLayout(layout)
-@tagName('')
-export default class NaviVisualizationsGoalGaugeComponent extends Component {
-  @service naviMetadata;
-  @service naviFormatter;
-
+export default class GoalGaugeVisualization extends Component<Args> {
   /**
    * @property {Array} - List of class names added to the gauge component
    */
-  @computed
+  @computed()
   get widgetClassNames() {
     return ['goal-gauge-widget', `${guidFor(this)}-goal-gauge-widget`];
   }
 
-  /**
-   * @property {Number} - value to be rendered on gauge
-   */
-  actualValue;
+  @computed('args.{model.firstObject.request.metricColumns.[],options.metricCid}')
+  get metric(): ColumnFragment {
+    const { request } = this.args.model?.firstObject || {};
+    const { metricCid } = this.args.options;
+    const metricColumn = request?.metricColumns.find(({ cid }) => cid === metricCid);
+    assert(`A metric column should exist with cid: ${metricCid}`, metricColumn);
+    return metricColumn;
+  }
 
+  @computed('metric', 'args.model.firstObject.response.rows.[]')
+  get actualValue(): number {
+    const { model } = this.args;
+    if (model) {
+      const { response } = model?.firstObject || {};
+      const firstRow = response?.rows?.[0] || {};
+      const { canonicalName } = this.metric;
+      let actualValue: number = firstRow[canonicalName] as number;
+      return actualValue;
```

Comment:
```suggestion
      return firstRow[canonicalName] as number;
```

Reply:
currently have this unpushed right now `return Number(firstRow[canonicalName]);`